cmd/buildkitd/main_windows.go

```diff
@@ -7,10 +7,36 @@ import (
 	"crypto/tls"
 	"net"
 
+	"github.com/Microsoft/go-winio"
 	_ "github.com/moby/buildkit/solver/llbsolver/ops"
 	"github.com/pkg/errors"
 )
 
+const (
+	defaultTraceSocketPath = `\\.\pipe\otel-grpc`
+)
+
 func listenFD(addr string, tlsConfig *tls.Config) (net.Listener, error) {
 	return nil, errors.New("listening server on fd not supported on windows")
 }
+
+func traceSocketPath(root string) string {
+	return defaultTraceSocketPath
+}
+
+func getLocalListener(listenerPath string) (net.Listener, error) {
+	pc := &winio.PipeConfig{
+		// Allow generic read and generic write access to authenticated users
+		// and system users. On Linux, this pipe seems to be given rw access to
+		// user, group and others (666).
+		// TODO(gabriel-samfira): should we restrict access to this pipe to just
+		// authenticated users? Or Administrators group?
+		SecurityDescriptor: "D:P(A;;GRGW;;;AU)(A;;GRGW;;;SY)",
```

Comment:
@helsaawy What do you think?

Reply:
I think the default permission is everyone can read, but only admins and the creator can write, which feels fine to me (at least, in containerd we dont increase permissions above that).
We can always increase the security permissions later (or add a protected prefix to the pipe path...)